pkg/dsdk/connection.go

```diff
@@ -356,8 +363,8 @@ func (c *ApiConnection) do(ctxt context.Context, method, url string, ro *greq.Re
 		if c.hasLoggedIn() {
 			c.Logout()
 			if apiresp, err2 := c.Login(ctxt); apiresp != nil || err2 != nil {
-				Log().Errorf("%s", err)
-				Log().Errorf("%s", err2)
+				detailLog.Errorf("%s", err)
+				detailLog.Errorf("%s", err2)
```

Comment:
`err` must be `badStatus[PermissionDenied]` here so we don't need the first log event.  Wouldn't hurt to add a little extra context around the second.
```suggestion
				detailLog.Errorf("failed to re-authenticate before retrying request: %s", err2)
```